model/system_test.go

```diff
@@ -35,23 +35,17 @@ func TestSystemTransformation(t *testing.T) {
 	nodename, podname, podUID := "a.node", "a.pod", "b.podID"
 
 	for name, system := range map[string]System{
-		"hostname":                               System{DetectedHostname: detected},
-		"ignored hostname":                       System{ConfiguredHostname: configured},
-		"full hostname info":                     System{DetectedHostname: detected, ConfiguredHostname: configured},
-		"k8s nodename with hostname":             System{Kubernetes: Kubernetes{NodeName: nodename}, DetectedHostname: detected},
-		"k8s nodename with configured hostname":  System{Kubernetes: Kubernetes{NodeName: nodename}, ConfiguredHostname: configured},
-		"k8s podname":                            System{Kubernetes: Kubernetes{PodName: podname}, DetectedHostname: detected},
-		"k8s podUID":                             System{Kubernetes: Kubernetes{PodUID: podUID}, DetectedHostname: detected},
-		"k8s namespace":                          System{Kubernetes: Kubernetes{Namespace: namespace}, DetectedHostname: detected},
-		"k8s podname with configured hostname":   System{Kubernetes: Kubernetes{PodName: podname}, DetectedHostname: detected, ConfiguredHostname: configured},
-		"k8s podUID with configured hostname":    System{Kubernetes: Kubernetes{PodUID: podUID}, DetectedHostname: detected, ConfiguredHostname: configured},
-		"k8s namespace with configured hostname": System{Kubernetes: Kubernetes{Namespace: namespace}, DetectedHostname: detected, ConfiguredHostname: configured},
-		"k8s empty":                              System{Kubernetes: Kubernetes{}, DetectedHostname: detected, ConfiguredHostname: configured},
```

Comment:
Kubernetes is generally covered by kubernetes_test.go, and the logic for setting `host.hostname` for events with kubernetes metadata is covered by the modeprocessor. The tests are redundant now.